### ISSUE TITLE
resource count cannot reference resource variable

### ISSUE THREAD

**Issue by gtmtech** [hashicorp/terraform#12113]
terraform 0.8.7

Getting this error:

`* aws_vpc_peering_connection_accepter.platform_prod: resource count can't reference resource variable: data.aws_caller_identity.current.id`

I cant think of any reason why this should not be allowed:

```
resource "aws_vpc_peering_connection_accepter" "platform_prod" {
    count       = "${var.flags["enable_peer_platform_prod"] * (data.aws_caller_identity.current.id == var.account_ids["prod"] ? 0 : 1)}"
    provider    = "aws.prod"
    auto_accept = "${data.aws_caller_identity.current.id == var.account_ids["prod"] ? false : true}"
  
    tags {
        Name = "${var.project}_${var.environment}_to_platform_prod"
    }
}
```

**Comment by fabian-dev** [comment]
Using `count` on a data source is solved for TF 0.9. See #11482

**Comment by mitchellh** [comment]
Thanks @fabian-dev. Done!
